### PR TITLE
Ice-Water Classification from Sentinel-1 SAR imagery

### PR DESCRIPTION
**Pull request by CatherineTaelman**
__IMPORTANT:__ _Check that none of your commits has hard-coded credentials (user, password)! Otherwise, don't open a pull request and get in touch with us first._

_INFO: The checklist below will be edited by the CVL team. If you want to add any description use the place holder in the end.
To update the pull request just commit to the related branch. There is no need to delete the pull request and reopen a new one._
# CVL submission criteria

## Structure
- [x] Introduction
- [x] Load data
- [x] Process data
- [x] Result visualisation
- [x] Summary of notebook outcomes

## Content requirements
- Self-contained
  - [x] Runs successfully
  - [x] Reproducible
- [x] Sufficient inline (markdown) documentation
- [x] In accordance with project description
- [x] Utilisation of CVL data catalogue (if applicable, alternative portals such as Zenodo could be utilised as well)

---
# Description of pull request
_< your description here (optional) >_

### CODE REVIEW

**Comment by ykern** [comment]
_Issue from email conversation:_
> In my script `load_and_calibrate_S1_scene`, I load a S1 image in netCDF format and then perform radiometric calibration using the LUT provided. This worked fine when I wrote this code some weeks ago, but when I try to run it now, the server times out after trying to run that cell for 10 (?) minutes. See error message in the screenshot.
> 
> I originally planned to stitch together this `load_and_calibrate_S1_scene` script with the `ice_water_classification` script, because the latter takes as input the calibrated bands that are produced in the former.
> For now, I have left both scripts as separate files, and I provide the input data for the classification part in the folder called `data` (which I produce locally, and then upload to the notebook).
> 
> It would however be neat to stitch the two scripts together, and in that way use the data from the CVL data catalogue for the classification part too (instead of producing that input locally).
> 
> Screenshot of issue:
> Screenshot from 2023-02-06 22-36-39

**Comment by ykern** [comment]
Amazing work and very nice inline markdown documentation! Well done! :+1: 
The issue mentioned above from `load_and_calibrate_S1_scene` does not occur for me when running all your notebooks in JupyterHub. So I assume that this should be fine.

Otherwise, everything runs fine. Only the second last step
```python
# predict label for each pixel
predicted_image = lp_model.predict(X)
# reshape to 2D image
predicted_image = np.reshape(predicted_image, (label_mask.shape))
```
raises the following `RuntimeWarning`
```bash
../site-packages/sklearn/semi_supervised/_label_propagation.py:231: RuntimeWarning: invalid value encountered in divide
  probabilities /= normalizer
```
However, the code afterwards seems like it still works and I see that you also had the same warning in the `ice_water_classification.ipynb` in your repo. So assume this is nothing critical?

### data
~~One question about understanding the content of the `data` directory. I see the files are loaded in `ice_water_classification.ipynb`. I tried to understand where they come from. Are those files programmatically generated? Can these be / Are these generated via the notebook?~~
Edit: Nevermind, I didn't see the top markdown cell which explains where those files are coming from.

**Comment by ykern** [comment]
Can you run the following command in terminal (while rooted at the top level of the repo) to recover the deleted `RENAME-your-project/` directory
```
git checkout 1bd8d00bc7cef22d08d589765bd9c6044bee7f7f -- RENAME-your-project/README.md
```
and then commit again to the pull request?
Otherwise, your pull request asks for deletion of that directory.
I think that would be the last missing piece before we can merge :+1:

**Comment by CatherineTaelman** [comment]
Thank you for the feedback Yannick! 

**Regarding the RunTimeWarning message**: this seems to be a bug in the Label Propagation of sklearn (see this discussion: https://github.com/scikit-learn/scikit-learn/issues/9292). One of the 'tricks' to get rid of this warning seems to be to increase the number of neighbours when building the graph. I'm trying this out right now, and will let you know if this fixes the warning. 
It does, however, not seem critical and the results are as expected.

**For the error with calibrating the full S1 scene**, it actually runs normally today (just tried it again). This means that I could actually now stitch together the _load_and_calibrate_S1_scene_ with _ice_water_classification_. I could still do this later in the week, would that be of any added value to the notebook?

**Comment by CatherineTaelman** [comment]
Update: runtimewarning is fixed! I've just pushed the updated code.

**Comment by ykern** [comment]
Was just about to send a comment that there is not need to look into the RuntimeWarning if it is not crucial, but I see you already fixed it anyway. Great job :+1: 

About the stitching of the scripts. I think you have done very good and extensive work on this project and the Readme is clear about how to run the notebooks. 
I will leave this up to you but it is definitely not required to spend more time on this. We are very happy with how it looks now! :-) I will merge it how it is and if you feel like you want to add any further updates just open a new pull request.

**Comment by CatherineTaelman** [comment]
Okay sounds good, I will let you know by tomorrow if I get some time to stitch the scripts together. Thanks Yannick:)

**Comment by CatherineTaelman** [comment]
Hei Yannick, I didn't have more time this week to look at stitching the scripts together, so I decided to leave it as is. Thank you for the help during this project!
